src/Builder/Meta.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MongoDB\Builder;
+
+use MongoDB\Builder\Type\Encode;
+use MongoDB\Builder\Type\ExpressionInterface;
+use MongoDB\Builder\Type\OperatorInterface;
+
+/**
+ * Returns the metadata associated with a document, e.g. "textScore" when performing text search.
+ *
+ * @see https://www.mongodb.com/docs/manual/reference/operator/aggregation/meta/
+ */
+enum Meta: string implements OperatorInterface, ExpressionInterface
```

Comment:
@alcaeus suggested to name it `SortMeta`. https://github.com/mongodb/mongo-php-builder/pull/56#discussion_r1467476295

Reply:
[$meta: Behavior](https://www.mongodb.com/docs/manual/reference/operator/aggregation/meta/#behavior) suggests the operator can be used in context beyond sorting, such as projections.

Reply:
The `ExpressionInterface` allows to use it anywhere an expression is expected.